indra/newview/llinventorymodel.cpp

```diff
@@ -451,17 +452,16 @@ void LLInventoryModel::getDirectDescendentsOf(const LLUUID& cat_id,
 	items = get_ptr_in_map(mParentChildItemTree, cat_id);
 }
 
-LLMD5 LLInventoryModel::hashDirectDescendentNames(const LLUUID& cat_id) const
+LLUUID LLInventoryModel::hashDirectDescendentNames(const LLUUID& cat_id) const
```

Comment:
I suggest using some alias instead of an LLUUID here to make sure it will be more obvious at first glance that this returns not an actual UUID.

Reply:
I could add a digest_t type to HBXXH* classes (i.e. 'typedef LLUUID digest_t' for HBXXH128, in this case), and change that method apparent signature for HBXXH128::digest_t 'LLInventoryModel::hashDirectDescendentNames(const LLUUID& cat_id) const', but then it would mean including the hbxxh.h header in llinventorymodel.h (meaning indirectly as well in a load of other modules including llinventorymodel.h)
Alternatively, digest_t could be made a typedef in LLInventoryModel only (avoiding the compile-time-costly hbxxh.h header inclusion)...

Reply:
I opted for the second (simpler) solution.

See the added commit.

Reply:
Looks good, thank you.

I was about to suggest just adding typedef into lluuid.h, while not the best place, it's the same thing so won't be too out of place.